modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DBBasedExpiryMessageDeletionTask.java

```diff
@@ -0,0 +1,169 @@
+             * This logic belongs to an MB run in stand alone mode / the coordinator node run in cluster mode
+             */
+            if (!isClusterEnabled || (isClusterEnabled && AndesContext.getInstance().getClusterAgent().isCoordinator())) {
+
+                Set<String> queues = andesContextStore.getAllQueues();
+
+                for (String queueName : queues) {
+                    //get the upper bound messageID for each unassigned slots as a set for the specific queue
+                    TreeSet<Long> messageIDSet = andesContextStore.getMessageIds(queueName);
+
+                    /**
+                     * deletion task run only if there are more messages than the safety slot count upper bound
+                     * otherwise deletion task is not run
+                     * minimum safetySlotCount is 1
+                     */
+                    if (safetySlotCount >= 1 && messageIDSet.size() >= safetySlotCount) {
+
+                        //set the lower bound Id for safety delete region as the safety slot count interval upper bound id + 1
+                        long currentDeletionRangeLowerBoundId = messageIDSet.
+                                toArray(new Long[messageIDSet.size()])[safetySlotCount - 1] + 1;
+
+                        //get the expired messages for that queue in the range of message ID starting form the lower bound ID
+                        List<AndesMessageMetadata> expiredMessages = MessagingEngine.getInstance()
+                                .getExpiredMessages(currentDeletionRangeLowerBoundId, queueName);
+
+                         /*
+                          * Checks for the message store availability if its not available
+                          * Deletion task needs to await until message store becomes available
+                         */
+                        if (messageStoresUnavailable != null) {
+
+                            log.info("Message store has become unavailable therefore expiry message deletion task" +
+                                    " waiting until store becomes available");
+                            //act as a barrier
+                            messageStoresUnavailable.get();
+                            log.info("Message store became available. Resuming expiry message deletion task");
+                            messageStoresUnavailable = null; // we are passing the blockade
+                            // (therefore clear the it).
+                        }
+
+                        if (null != expiredMessages && !expiredMessages.isEmpty()) {
+
+                            //in order to stop the slot delivery worker to this queue in the deletion range
+                            SlotManagerClusterMode.setCurrentDeletionQueue(queueName);
+                            SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+                            SlotManagerStandalone.setCurrentDeletionQueue(queueName);
```

Comment:
Depending on whther clustering is enabled or not, I guess we can remove the assignment to either 'SlotManagerClusterMode' or 'SlotManagerStandalone'. Can't we?